apps/src/music/views/MusicView.jsx

```diff
@@ -321,6 +322,21 @@ class UnconnectedMusicView extends React.Component {
     return this.props.isPlaying;
   };
 
+  getValidationTimeout = () => {
+    // The level can specify a desired timeout, in measures, when we can start showing
+    // non-success validation messages.  That said, if we've just completed playing the
+    // last measure before reaching that specified value, we can start showing the
+    // messages.
+    // If no timeout is specified, then we can starting showing the non-success messages
+    // at measure 2.
+    return this.props.levelData?.validationTimeout
+      ? Math.min(
+          this.props.levelData?.validationTimeout,
+          this.sequencer.getLastMeasure()
```

Comment:
One thought - why not flip this, and only check validation once we've completed all measures? That way we'll always wait until all sounds have played before evaluating? 

Thinking specifically about the case you showed in your video with the longer sound ("relaxed_beat") where the timeout is 6 measures - looks like we still show the failing message after 6 measures even though the student has queued up 3 sounds, which similarly confusing to what we currently have. Would it be better to instead just wait until the last measure in that case? And instead of being a validation "timeout" this could be more of a validation "delay", i.e. "wait at least this long before evaluating". This could be helpful for the trigger level where initially the total number of measures will be very short (or 0) but by providing this delay, we'd still give the student some time to trigger a sound before showing the failing message.

That being said, it might still be useful to have a timeout but I think it would work well to use it in conjunction with a delay as well. For example, if we had some validation around using a function, but the student used a repeat block to play a sound 100 times, it would still be good to have some timeout (ex. 16 measures).

Reply:
I think we could make this more elaborate, but we'll always have the tension between waiting a while but also not waiting too long.  We have a tradition (for better or worse) of timeouts in other labs (usually with countdown bar or "clock" UI elements) and while they aren't perfect, I think they make sense because we want to give users timely feedback, even if their program run for much longer.  You're very right that we would still need a timeout, even with a delay.  I do like that here, unlike in those other labs, the program keeps running, even after we've given that (hopefully) timely feedback.

So, I'm inclined to try just adding this single parameter for now, but we can see what we learn as we (and curriculum) use it.

Reply:
(Discussed offline) that sounds good to me. Happy to keep iterating on this as we add more levels and validation types as well!